app/src/org/commcare/utils/FileUtils.java

```diff
@@ -1,364 +0,0 @@
-        if (scaledBitmap != null) {
-            // Write this scaled bitmap to the final file location
-            FileOutputStream out = null;
-            try {
-                out = new FileOutputStream(finalFilePath);
-                scaledBitmap.compress(type.getCompressFormat(), 100, out);
-                return true;
-            } catch (Exception e) {
-                e.printStackTrace();
-                return false;
-            } finally {
-                try {
-                    if (out != null) {
-                        out.close();
-                    }
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
-        }
-        return false;
-    }
-
-    /**
-     * Attempts to scale down an image file based on the max dimension given, using the following
-     * logic: If at least one of the dimensions of the original image exceeds the max dimension
-     * given, then make the larger side's dimension equal to the max dimension, and scale down the
-     * smaller side such that the original aspect ratio is maintained.
-     *
-     * @param maxDimen - the largest dimension that we want either side of the image to have
-     * @return A scaled down bitmap, or null if no scale-down is needed
-     */
-    private static Bitmap getBitmapScaledByMaxDimen(Bitmap originalBitmap, int maxDimen) {
-        if (originalBitmap == null) {
-            return null;
-        }
-        int height = originalBitmap.getHeight();
-        int width = originalBitmap.getWidth();
-        int sideToScale = Math.max(height, width);
-        int otherSide = Math.min(height, width);
-
-        if (sideToScale > maxDimen) {
-            // If the larger side exceeds our max dimension, scale down accordingly
-            double aspectRatio = ((double)otherSide) / sideToScale;
-            sideToScale = maxDimen;
-            otherSide = (int)Math.floor(maxDimen * aspectRatio);
-            if (width > height) {
-                // if width was the side that got scaled
-                return Bitmap.createScaledBitmap(originalBitmap, sideToScale, otherSide, false);
-            } else {
-                return Bitmap.createScaledBitmap(originalBitmap, otherSide, sideToScale, false);
-            }
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * Copies from sourceFile to destFile (either a directory, or a path
-     * to the new file)
-     *
-     * @param sourceFile A file pointer to a file on the file system
-     * @param destFile   Either a file or directory. If a directory, the
-     *                   file name will be taken from the source file
-     */
-    public static void copyFile(File sourceFile, File destFile) {
-        if (sourceFile.exists()) {
-            if (destFile.isDirectory()) {
-                destFile = new File(destFile, sourceFile.getName());
-            }
```

Comment:
`FileUtil.copyFile` doesn't exhibit this behavior of allowing the destFile to be a directory. Looks like everywhere that calls this passes in a file, so it should be fine, except for in `DrawActivity` where it is unclear whether a dir or file is passed in.
I would leave this implementation in for simplicity and since it might be _slightly_ more efficient to use built-in channels instead of the hand-rolled stream writing in `AndroidStreamUtils`.